File: 프로그래머스/lv2/12953. N개의 최소공배수/N개의 최소공배수.py
def gcd(a,b):
    if b > a:
        i = a
        a = b
        b = i
    while b != 0:
        n = a % b
        a = b
        b = n
    return a

def solution(arr):
    answer = arr[0]
    for i in arr[1:]:
        answer = i * answer // gcd(answer, i)
    return answer

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.java
class Solution {
    public int solution(int[] arr) {
        int answer = arr[0] * arr[1] / getNum(arr[0], arr[1]);
        for(int i = 2; i < arr.length; i++) {
            answer = answer * arr[i] / getNum(answer, arr[i]);
        }
        return answer;
    }

    public int getNum(int a, int b) {
        while (b != 0) {
            int r = a % b;
            a = b;
            b = r;
        }
        return a;
    }
}